packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/internal_utils.ts

```diff
@@ -272,3 +274,30 @@ export function normalizeNamespace(namespace?: string) {
 export function getCurrentTime() {
   return new Date(Date.now()).toISOString();
 }
+
+/**
+ * Returns the managed boolean to apply to a document as it's managed value.
+ * For use by applications to modify behavior for managed saved objects.
+ * The behavior is as follows:
+ * If `optionsManaged` is set, it will override any existing `managed` value in all the documents being created
+ * If `optionsManaged` is not provided, then the documents are created with whatever may be assigned to their `managed` property
+ * or default to `false`.
+ *
+ * @internal
+ */
+
+export function setManaged({
```

Comment:
I understand the rationale for extracting this functionality to separate unit that can be tested, but in this case keeping the implementation specifics inside `create` and `bulkCreate` contexts is simpler. I think your repository test coverage gives strong guarantees about behaviour.

Reply:
Yeah, I agree that the few calls to `setManaged` could easily be replaced by inlining `optionsManaged ?? objectManaged ?? false`.

Now If we think it's safer to keep the util, I would be fine with it.

Reply:
Keeping the util means that we only need to make a change in one place, rather than "hunt down" all the places we use short-hand.